share/mk/src.opts.mk

```diff
@@ -268,8 +268,10 @@ __DEFAULT_DEPENDENT_OPTIONS+= ${var}_SUPPORT/${var}
 # overriden by bsd.compat.mk.
 .if defined(TARGET_ARCH) && !defined(WANT_COMPAT)
 __T=${TARGET_ARCH}
+__C=${TARGET_CPUTYPE}
 .else
 __T=${MACHINE_ARCH}
+__C=${MACHINE_CPUTYPE}
```

Comment:
Do we actually use MACHINE_CPUTYPE? a quick grep seems to indicate that `TARGET_CPUTYPE` sets `CPUTYPE`.
In `share/mk/bsd.cpu.mk` we set `cheri` as part of `MACHINE_CPU` but not `CPUTYPE`.
I'm not sure what the

It also seems like we use `MACHINE_CPU:Mcheri` in some places, maybe those should be `${CPUTYPE} == cheri`? Or should we always ensure that both variables are set?

Reply:
We can use MACHINE_CPU:Mcheri most everywhere and should do so.  It's only this location where we can't because it isn't set when src.opts.mk is included in Makefile.inc1.